src/sentry/static/sentry/app/components/confirm.jsx

```diff
@@ -7,14 +7,21 @@ import {t} from '../locale';
 
 class Confirm extends React.PureComponent {
   static propTypes = {
-    disabled: PropTypes.bool,
-    message: PropTypes.node.isRequired,
     onConfirm: PropTypes.func.isRequired,
-    onConfirming: PropTypes.func,
-    onCancel: PropTypes.func,
-    priority: PropTypes.oneOf(['primary', 'danger']).isRequired,
     confirmText: PropTypes.string.isRequired,
     cancelText: PropTypes.string.isRequired,
+    priority: PropTypes.oneOf(['primary', 'danger']).isRequired,
+    message: PropTypes.node,
+    /**
+     * Renderer that passes:
+     * `doConfirm`: Allows renderer to perform confirm action
+     * `doClose`: Allows renderer to toggle confirm modal
+     */
+    renderMessage: PropTypes.func,
```

Comment:
What do you think of `renderMessage((confirm, close) => {...})`? It seems a bit more straightforward to me.

Reply:
Yeah I like it, I was just looking for a convention for callback props that have side effects when called.